makefile

```diff
@@ -1,10 +1,10 @@
 run: main
```

Comment:
Use variables.
compiler, compilation flags, linking flags and etc.

Reply:
Set the flags -Werror -Wall -g -fsanitize=address,undefined

The -Wall flag allows you to perform additional checks on the source code, displaying warnings.
The -Werror flag prevents a file from being compiled successfully if there are warnings (that is, all warnings become errors)
The -g -fsanitize=address,undefined flags will help you find errors when working with memory when running the program. However, for a release build, you generally don't need them.

Reply:
Phony Targets: Targets like run, clean and etc., should be marked as phony using the .PHONY